Name: Launcher: Reset to First Page
Version: 1.3.1-27
Author: Ian Grant
Description: Resets the launcher when closing so that it always opens to the first page, rather than the last page used. It also scrolls each page back to the top.  (MIT license)

--- /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js.orig	Sat Dec  5 11:48:12 2009
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js	Thu Dec  3 19:42:03 2009
@@ -183,6 +183,19 @@
 		delete this.launchRequest;
 		
 		this.removeLaunchFeedback();
+		
+		/* Reset launcher to first page - seems we need to do scrollTo AND setSnapIndex to be fully compatible with Wrap Pages */
+		$('launcher_root').mojo.scrollTo(0, undefined, false, true);
+		this.activePageIndex = Math.abs($('launcher_root').mojo.getScrollPosition().left) / this.kPageWidth;
+		$('launcher_root').mojo.setSnapIndex(0, false);
+		this.updatePageIndicators();
+		
+		/* Jump scroll position back to top for all pages */
+		var numPages=this.pagesModel.getNumPages();
+		for (var i=0; i<numPages; i++) {
+			var scroller = this.getPageScroller(i);
+			if (scroller && scroller.mojo) { scroller.mojo.revealTop(); }
+		}
 	},
 	
 	onResize: function(event) {
